### PR TITLE
Four bugfixes

### PR DESCRIPTION
**Pull request by dkreuter**
pls review kthxbai

### CODE REVIEW

File: libr/util/p_format.c [hunk 1]
```diff
@@ -1058,6 +1059,8 @@ R_API int r_print_format(RPrint *p, ut64 seek, const ut8* b, const int len,
 					dot = strchr (field, '.');
 				if (dot)
 					*dot = '\0';
+				if (oarg != NULL)
```
Comment (jjdredd) [inline libr/util/p_format.c]: why?

**Comment by jjdredd** [comment]
also, setting a pointer after a 'free' to NULL , when it isn't going out of the scope helps to avoid double free or use after free.

**Comment by radare** [comment]
Use R_FREE() for short if u like

**Comment by jjdredd** [comment]
oh nice macro

**Comment by radare** [comment]
anyway.. a free before a return doesnt needs to be nullified.

**Pull request closed without merging**